policies/documentation-policy.md

```diff
@@ -0,0 +1,91 @@
+# OpenSSL documentation policy
+
+This document describes the code documentation and commenting
+requirements for the OpenSSL project.
+
+The project's documentation is all about making the libraries and tools
+more accessible to our users.
+
+## Chapter 1: Command line commands and arguments
+
+All new or modified arguments to the commands must be documented in the
+`doc/man1` directory.  This documentation is in _POD_ format.
+
+All new commands must be documented in the `doc/man1` directory.  This
+documentation is in _POD_ format.
+
+## Chapter 2: Public symbols in the libraries
+
+All new public symbols must be documented in a _POD_ manual page in the
+`doc/man3` directory.  This includes both macros and functions.
+
+The allowed exceptions are:
+
+- guard macros preventing a header file being included twice
+- new OPENSSL_NO_DEPRECATED_ macros
+- new symbols generated automatically via `make update` (errors, objects, etc)
+
+## Chapter 3: Internal functions, structures, globals and macros
+
+Internal functions and macros are those defined in the `include/crypto`
+and `include/internal` directories.
+
+These should all be documented at the point of definition.  A comment
+describing their purpose and the input and output arguments for functions
+suffices.
+
+For _trivial_ items, where their operation is obvious from their
+definition, the documentation requirement is not required.  The following
+are generally representative of trivial items, however it is quite
+possible for any of these to be non-trivial in specific instances and
+therefore require documentation:
+
+- OSSL_DISPATCH tables
+- method structures
+- upref functions
+- free functions
+- simple getter/setter functions
```

Comment:
Where structures are opaque the "fields" of the underlying structure need to be documented and in the setters/getters is where that documentation will occur. Unless you plan to require documenting this somewhere else. So I would **not** include setter/getter in the list of normally not documented.

Reply:
Hmmmm, there are a number of _get0_name, _get_number, get_size where a comment isn't adding anything meaningful.  There are others where there should be some documentation.

Reply:
As I understand it this is in regards to internal documentation so the fields are not really opaque here and so it should be sufficient to document the structure members itself via comments. For public getters/setters the documentation is required.